### PR TITLE
CUMULUS-3077 Update data-migration2 to remove excess files

### PR DESCRIPTION
**Pull request by charleshuang80**
**Summary:** Summary of changes

Addresses [CUMULUS-3077: Data migration does not remove excess files](https://bugs.earthdata.nasa.gov/browse/CUMULUS-3077)

## Changes

* add removeExcessFiles method so files no longer associated with a granule are removed during migration
* add tests to cover various scenarios for the value for 'file' on a granule during migration including empty array and undefined
* add test to cover if values for file attributes are changed and the granule is re-migrated

## PR Checklist

- [x] Update CHANGELOG
- [x] Unit tests
- [ ] Ad-hoc testing - Deploy changes and test manually
- [ ] Integration tests

### CODE REVIEW

File: lambdas/data-migration2/tests/test-granulesAndFiles.js [hunk 1]
```diff
@@ -900,11 +900,343 @@ test.serial('migrateGranuleAndFilesViaTransaction processes granule with no file
...
+    }
+  );
+
+  t.teardown(async () => {
+    await t.context.granulePgModel.delete(t.context.knex, { cumulus_id: records[0].cumulus_id });
+    // shouldn't need to delete files???
+  });
+});
+
+test.only('migrateGranuleAndFilesViaTransaction updates previously migrated file records correctly if the files are modified and the granule is re-migrated', async (t) => {
```
Comment (jennyhliu) [inline lambdas/data-migration2/tests/test-granulesAndFiles.js:1146]:
```suggestion
test('migrateGranuleAndFilesViaTransaction updates previously migrated file records correctly if the files are modified and the granule is re-migrated', async (t) => {
```

File: lambdas/data-migration2/tests/test-granulesAndFiles.js [hunk 2]
```diff
@@ -900,11 +900,343 @@ test.serial('migrateGranuleAndFilesViaTransaction processes granule with no file
...
+      migrated: 2,
+    },
+    granulesResult: {
+      total_dynamo_db_records: 1,
+      failed: 0,
+      skipped: 0,
+      migrated: 1,
+    },
+  });
+
```
Comment (jennyhliu) [inline lambdas/data-migration2/tests/test-granulesAndFiles.js:1043]: Can we check pg record after the first migration?

File: lambdas/data-migration2/tests/test-granulesAndFiles.js [hunk 3]
```diff
@@ -900,11 +900,343 @@ test.serial('migrateGranuleAndFilesViaTransaction processes granule with no file
...
+  testGranule.files.push(fakeFile2);
+
+  // migrate 1st time
+  await migrateGranuleAndFilesViaTransaction({
+    dynamoRecord: testGranule,
+    knex,
+    loggingInterval: 1,
+  });
+
+  delete testFile1.size;
```
Comment (jennyhliu) [inline lambdas/data-migration2/tests/test-granulesAndFiles.js:1172]: suggest to use `modifiedTestFile1` `modifedFakeFile2`  etc. instead of update the original files. e.g.
const modifiedTestFile1 = omit(testFile1, ['size', ...])
const modifedFakeFile2 = {
...fakeFile2,
key: 'new key 2',
}

File: lambdas/data-migration2/tests/test-granulesAndFiles.js [hunk 4]
```diff
@@ -900,11 +900,343 @@ test.serial('migrateGranuleAndFilesViaTransaction processes granule with no file
...
+      skipped: 0,
+      migrated: 1,
+    },
+  });
+
+  t.deepEqual(
+    omit(fileRecords[0], fileOmitList),
+    {
+      bucket: testFile1.bucket,
+      checksum_value: null,
```
Comment (jennyhliu) [inline lambdas/data-migration2/tests/test-granulesAndFiles.js:1215]: Shouldn't be the `checksum_value` etc.  not in the record at all?

**Comment by jennyhliu** [comment]
Does the data-migration2 update Elasticsearch records?  Do we know if the Elasticsearch records are consistent with the records in DynamoDB?

File: lambdas/data-migration2/tests/test-granulesAndFiles.js (hunk 1 above)
Comment (charleshuang80) [reply lambdas/data-migration2/tests/test-granulesAndFiles.js]: changed

File: lambdas/data-migration2/tests/test-granulesAndFiles.js (hunk 2 above)
Comment (charleshuang80) [reply lambdas/data-migration2/tests/test-granulesAndFiles.js:1046]: added

File: lambdas/data-migration2/tests/test-granulesAndFiles.js (hunk 3 above)
Comment (charleshuang80) [reply lambdas/data-migration2/tests/test-granulesAndFiles.js]: updated

**Comment by charleshuang80** [comment]
In the test file I was working in, I did not see anything testing Elasticsearch. I will look around in the lambda files to see if there is anything relating to Elasticsearch I can or should use/test.

**Review by jennyhliu — APPROVED** [review]
👍
